main/src/library/File.flix

```diff
@@ -485,14 +495,23 @@ namespace File {
             import java.io.FileInputStream.read(Array[Int8, Static], Int32, Int32): Int32 & Impure;
 
             let javaFile = newFile(f);
-            let stream = newInputStream(javaFile);
-            discard skip(stream, opts.offset);
-            let bytes = [0i8; opts.count];
-            let numberRead = read(stream, bytes, 0, opts.count);
-
-            let res = readBytesWithHelper(bytes, numberRead, opts.count);
-
-            Ok(res)
+            def loop(i) = {
+                if (0 <= i) {
+                    let stream = newInputStream(javaFile);
+                    let skipped = skip(stream, opts.offset);
+                    if (skipped == opts.offset) {
+                        let bytes = [0i8; opts.count];
+                        let numberRead = read(stream, bytes, 0, opts.count);
+                        let res = readBytesWithHelper(bytes, numberRead, opts.count);
+                        Ok(res)
+                    } else {
```

Comment:
Given that this loop-skip is used more than once it should probably be its own `def skip(f, bytes): Result` kind of thing